usaspending_api/references/v2/views/agency.py

```diff
@@ -40,20 +33,37 @@ def get(self, request, pk, format=None):
             return Response(response)
 
         toptier_agency = agency.toptier_agency
-        # get corresponding submissions through cgac code
-        queryset = SubmissionAttributes.objects.all()
-        queryset = queryset.filter(
-            toptier_code=toptier_agency.toptier_code, submission_window__submission_reveal_date__lte=now()
+
+        # Get corresponding submissions through cgac code with Submission in latest Quarter / Period
+        submission_queryset = SubmissionAttributes.objects.filter(
+            toptier_code=toptier_agency.toptier_code,
+            submission_window__submission_reveal_date__lte=now(),
         )
 
-        # get the most up to date fy and quarter
-        queryset = queryset.order_by("-reporting_fiscal_year", "-reporting_fiscal_quarter")
-        queryset = queryset.annotate(
-            fiscal_year=F("reporting_fiscal_year"), fiscal_quarter=F("reporting_fiscal_quarter")
+        if not submission_queryset.exists():
+            return Response(response)
+
+        most_recent_quarter_window_id = get_last_closed_submission_date(is_quarter=True)["id"]
+        most_recent_period_window_id = get_last_closed_submission_date(is_quarter=False)["id"]
+
+        submission = (
+            submission_queryset.filter(
+                Q(submission_window=most_recent_quarter_window_id) | Q(submission_window=most_recent_period_window_id)
+            )
+            .order_by("-reporting_fiscal_year", "-reporting_fiscal_quarter", "-reporting_fiscal_period")
+            .annotate(fiscal_year=F("reporting_fiscal_year"), fiscal_quarter=F("reporting_fiscal_quarter"))
+            .first()
         )
-        submission = queryset.first()
         if submission is None:
-            return Response(response)
+            # If there is no Submission for the latest Quarter / Period then get Fiscal Date from latest submission;
+            # Not terminating early as we do above since the Agency does have at least one historical submission
+            submission = (
+                SubmissionAttributes.objects.filter(submission_window__submission_reveal_date__lte=now())
+                .order_by("-reporting_fiscal_year", "-reporting_fiscal_quarter", "-reporting_fiscal_period")
+                .annotate(fiscal_year=F("reporting_fiscal_year"), fiscal_quarter=F("reporting_fiscal_quarter"))
+                .first()
+            )
```

Comment:
🤦 
> Right? Or did I miss something.

No, you are right. Definitely overlooked this lower part / wasn't thinking about it right. Will get a fix out for this.

Reply:
Actually, just remembered the logic behind this and added a comment to explain. The thought is that this `SubmissionAttributes` check will get the latest Fiscal Year and Fiscal Period to populate the `current_total_budget_authority_amount`, but since there is no Submission for the Agency the values pulled from `AppropriationAccountBalances` will all be zero for that FY, FQ, and FP.